carpics_360.js

```diff
@@ -0,0 +1,540 @@
+            var autospin;
+            if (typeof givenDirection == "undefined") {
+                direction = this.direction;
+                autospin = true;
+            } else {
+                direction = givenDirection;
+                autospin = false;
+            }
+            var nextImage;
+            if (!this.spinStatus && autospin) {
+                return;
+            }
+            if (direction == 1) {
+                nextImage = this.CurrentImage.NextImage
+            } else if (direction == -1) {
+                nextImage = this.CurrentImage.PreviousImage;
+            } else {
+                return;
+            }
+            for (var i = 0; i < this.AllImages.length; i++) {
+                if (!nextImage.isReady) {
+                    if (direction == 1) {
+                        nextImage = nextImage.NextImage
+                    } else if (direction == -1) {
+                        nextImage = nextImage.PreviousImage;
+                    }
+                    continue;
+                } else {
+                    break;
+                }
+            }
+            if (!nextImage.isReady) {
+                return;
+            }
+            var previous = this.CurrentImage;
+            this.CurrentImage = nextImage;
+            this.CurrentImage.HTMLElement.style.display = "block";
+            previous.HTMLElement.style.display = "none";
+
+        }
+
+
+        if (!this.mouseDisabled) {
+            $(document).on("mousedown", "#" + this.divId, (function(thisObj) {
+                return function(baseEvent) {
+                    baseEvent.preventDefault();
+                    var thisTouch = Date.now();
+
+                    if (thisObj.zoomed || baseEvent.button === 2) {
+                        return;
+                    }
+                    thisObj.spinStatus = false;
+                    thisObj.turnStatus = true;
+                    thisObj.mouseXPosition = baseEvent.pageX;
+                    thisObj.mouseYPosition = baseEvent.pageY;
+                    thisObj.spinnerDiv.style.cursor = "-webkit-move";
+                    thisObj.spinnerDiv.style.cursor = "move";
+                    thisObj.spinnerDiv.style.cursor = "-moz-move";
+                }
+            })(this))
+
+            $(document).on("dblclick", "#" + this.divId, (function(thisObj) {
+                return function(baseEvent) {
+                    baseEvent.preventDefault();
+                    var releaseMouse = thisObj.zoomToggle(baseEvent);
+                    if (thisObj.zoomed === true) {
+                        thisObj.spinStatus = false;
+                        thisObj.turnStatus = false;
+                    } else {
+                        thisObj.spinStatus = thisObj.spinDefault;
+                        thisObj.turnStatus = false;
+                    }
+
+                }
+            })(this))
+            $(document).on("taphold", "#" + this.divId, (function(thisObj) {
```

Comment:
add empty line. make this consistent. anything with `$(document).on` will have one before